### PR TITLE
Adding Pinot Minion client

### PR DESCRIPTION
**Pull request by fx19880617**
## Description
Adding Pinot Minion client to operate on controller /tasks APIs.

### CODE REVIEW

File: pinot-common/src/test/java/org/apache/pinot/common/minion/MinionClientTest.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.InetSocketAddress;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+
+public class MinionClientTest {
```
Comment (Jackie-Jiang) [inline pinot-common/src/test/java/org/apache/pinot/common/minion/MinionClientTest.java:32]: This test is not actually testing the functionality as the response is hard-coded within the test. You may modify the minion integration test to test against the controller.

**Review by Jackie-Jiang — APPROVED** [review]
LGTM otherwise.
Minion is not ready to be used out of the box, and the APIs are subject to be changed. Not sure if we should add the client now, or hold until minion is ready

File: pinot-common/src/test/java/org/apache/pinot/common/minion/MinionClientTest.java (hunk 1 above)
Comment (fx19880617) [reply pinot-common/src/test/java/org/apache/pinot/common/minion/MinionClientTest.java:32]: will do

**Comment by fx19880617** [comment]
> LGTM otherwise.
> Minion is not ready to be used out of the box, and the APIs are subject to be changed. Not sure if we should add the client now, or hold until minion is ready

this is a good question, I will comment and make it a test feature I guess. I feel it would be good to have it and it's ok to change it and deprecate the methods in the future.

**Pull request merged**